src/HttpDriver/Driver.php

```diff
@@ -62,16 +67,16 @@ public function getUri()
     private function getHttpClient()
     {
         $options = [];
-        if (null !== $this->config->getTimeout()) {
-            $options['timeout'] = $this->config->getTimeout();
+        if ($this->config->hasValue('timeout')) {
+            $options['timeout'] = $this->config->getValue('timeout');
         }
 
-        if (null !== $this->config->getCurlInterface()) {
-            $options['curl'][10062] = $this->config->getCurlInterface();
+        if ($this->config->hasValue('curl_interface')) {
+            $options['curl'][10062] = $this->config->getValue('curl_interface');
         }
 
         if (empty($options)) {
-            return $this->config->getHttpClient();
+            return $this->config->get('http_client');
```

Comment:
I think we shouldn't care. I expect people having to modify the client to know what they're doing. In 95% of the use cases you would never touch the underlying transport.

But I have 5% of use cases where people need to do it :)